commands.c

```diff
@@ -0,0 +1,167 @@
+            { "zrank",                  CMD_SPEC_READONLY },
+            { "zrevrank",               CMD_SPEC_READONLY },
+            { "zscan",                  CMD_SPEC_READONLY },
+            { "hmget",                  CMD_SPEC_READONLY },
+            { "hlen",                   CMD_SPEC_READONLY },
+            { "hstrlen",                CMD_SPEC_READONLY },
+            { "hkeys",                  CMD_SPEC_READONLY },
+            { "hvals",                  CMD_SPEC_READONLY },
+            { "hgetall",                CMD_SPEC_READONLY },
+            { "hexists",                CMD_SPEC_READONLY },
+            { "hscan",                  CMD_SPEC_READONLY },
+            { "randomkey",              CMD_SPEC_READONLY },
+            { "keys",                   CMD_SPEC_READONLY },
+            { "scan",                   CMD_SPEC_READONLY },
+            { "dbsize",                 CMD_SPEC_READONLY },
+            { "ttl",                    CMD_SPEC_READONLY },
+            { "bitcount",               CMD_SPEC_READONLY },
+            { "georadius_ro",           CMD_SPEC_READONLY },
+            { "georadiusbymember_ro",   CMD_SPEC_READONLY },
+            { "geohash",                CMD_SPEC_READONLY },
+            { "geopos",                 CMD_SPEC_READONLY },
+            { "geodist",                CMD_SPEC_READONLY },
+            { "pfcount",                CMD_SPEC_READONLY },
+            { "sync",                   CMD_SPEC_UNSUPPORTED },
+            { "psync",                  CMD_SPEC_UNSUPPORTED },
+            { "reset",                  CMD_SPEC_UNSUPPORTED },
+            { "bgrewriteaof",           CMD_SPEC_UNSUPPORTED },
+            { "slaveof",                CMD_SPEC_UNSUPPORTED },
+            { "replicaof",              CMD_SPEC_UNSUPPORTED },
+            { "debug",                  CMD_SPEC_UNSUPPORTED },
+            { "auth",                   CMD_SPEC_DONT_INTERCEPT },
+            { "ping",                   CMD_SPEC_DONT_INTERCEPT },
+            { "save",                   CMD_SPEC_DONT_INTERCEPT },
+            { "bgsave",                 CMD_SPEC_DONT_INTERCEPT },
+            { "module",                 CMD_SPEC_DONT_INTERCEPT },
+            { "raft",                   CMD_SPEC_DONT_INTERCEPT },
+            { "raft.entry",             CMD_SPEC_DONT_INTERCEPT },
+            { "raft.config",            CMD_SPEC_DONT_INTERCEPT },
+            { "raft.cluster",           CMD_SPEC_DONT_INTERCEPT },
+            { "raft.shardgroup",        CMD_SPEC_DONT_INTERCEPT },
+            { "raft.node",              CMD_SPEC_DONT_INTERCEPT },
+            { "raft.ae",                CMD_SPEC_DONT_INTERCEPT },
+            { "raft.requestvote",       CMD_SPEC_DONT_INTERCEPT },
+            { "raft.loadsnapshot",      CMD_SPEC_DONT_INTERCEPT },
+            { "raft.debug",             CMD_SPEC_DONT_INTERCEPT },
+            { "raft.info",              CMD_SPEC_DONT_INTERCEPT },
```

Comment:
my personal asthetic doesn't tink the raft commands should be intermixed with the regular redis commands here.  while they are mostly their own block, it could make it a little hard to make sure all redis commands are covered?

Reply:
also, should all write command be enumerated here (i.e. essentialy to enumerate all commands not assume something not listed here is not a write command)